requirements.txt

```diff
@@ -1,6 +1,7 @@
 torch>=1.9
+numpy<2
 pydot
-git+https://github.com/bdusell/semiring-einsum
+torch-semiring-einsum>=1.2
```

Comment:
Yes, my problem was that I had an outdated version installed, as you pointed out. But I also thought it would be better to track a released version instead of the bleeding edge.